sycl/doc/extensions/proposed/sycl_ext_intel_fpga_datapath.asciidoc

```diff
@@ -0,0 +1,239 @@
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FPGA_DATAPATH` to one of the values defined in the table
+below. Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Representation of FPGA memory
+
+[source,c++]
+----
+namespace sycl::ext::intel::experimental {
+template <typename T>
+class fpga_datapath {
+  ...
+----
+
+`fpga_datapath` is a class template, parameterized by the type of the underlying
+allocation _T_ which encodes the size of the allocation.
+
+`fpga_datapath` initializes the underlying object of type `T` with the values passed into its constructor.
+
+[NOTE]
+====
+
+`fpga_datapath` has similar semantics to a reference wrapper. An fpga_datapath templated on a type T can implicitly convert to T, thereby allowing it to be used in most places that an object of type T could be used. The dot operator
+(`operator.`) cannot be overloaded, so a `get()` member is provided to allow a
+reference to be extracted directly when needed.
+====
+
+=== FPGA Datapath Interface
+
+The section below and the table following describe the constructors and member functions for `fpga_datapath`.
+
+[source,c++]
+----
+namespace sycl::ext::intel::experimental {
+
+  template <typename T>
+  class fpga_datapath {
+  public:
+    template <typename... S> constexpr fpga_datapath(S... args);
+    
+    // Access the underlying data
+    operator T &() noexcept;
+    operator const T &() const noexcept;
+
+    T &get() noexcept;
+    const T &get() const noexcept;
+
+    // Enable assignments from underlying type
+    fpga_datapath &operator=(const T &) noexcept;
+
+    // Note that there is no need for "fpga_datapath" to define member functions
+    // for operators like "++", "[]", "->", comparison, etc. Instead, the type
+    // "T" need only define these operators as non-member functions. Because
+    // there is an implicit conversion from "fpga_datapath" to "T&", the operations
+    // can be applied to objects of type "fpga_datapath<T>"
+  };
+
+} // namespace sycl::ext::intel::experimental
+----
+
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template<typename ... S>
+constexpr fpga_datapath(S ... args);
+----
+|
+Constructs a `fpga_datapath` object, and implicit storage for `T`.
```

Comment:
It's worth mentioning that this calls T's constructor with arguments "args..." to initialize the underlying data.